csdmpy/numpy_wrapper/__init__.py

```diff
@@ -37,19 +37,39 @@ def fft(csdm, axis=0):
     ndim = len(csdm_new.dimensions)
 
     if dimension_object._complex_fft:
+        scale_factor = 1.0 if np.isfinite(dimension_object.reciprocal.period) else 2.0
         phase = np.exp(2j * np.pi * coordinates_offset_res * coordinates)
         phase_grid = _get_broadcast_shape(phase, ndim, axis=index)
         for item in csdm_new.dependent_variables:
+            comp_shape_len = len(item.subtype._components.shape)
+            slice_ = tuple(
+                [
+                    slice(None, 1, None) if i == -index - 1 else slice(None)
+                    for i in range(comp_shape_len)
+                ][::-1]
+            )
```

Comment:
This code matches lines 64-71. Possible to create helper function to reduce code overlap?

```
def fft():
    """docstring"""
    def _get_slice(comp_shape_len, index):
        return tuple(
                [
                    slice(None, 1, None) if i == -index - 1 else slice(None)
                    for i in range(comp_shape_len)
                ][::-1]
            )

    # Code continues ...

    for item in csdm_new.dependent_variables:
            comp_shape_len = len(item.subtype._components.shape)
            _slice = _get_slice(comp_shape_len, index)

    # Code continues ...
```

Reply:
Addressed the comment